lib/rules/object-curly-newline.js

```diff
@@ -144,6 +144,9 @@ module.exports = {
                 )
             );
 
+            const allTokens = sourceCode.getTokens(node, { includeComments: true });
+            const hasComments = allTokens.some(token => astUtils.isCommentToken(token));
```

Comment:
Oops, sorry about that. Does this work?

```js
const hasComments = sourceCode.commentsExistBetween(
    sourceCode.getFirstToken(node),
    sourceCode.getLastToken(node)
);
```

The reason I bring this up is because I'm a bit concerned about the performance impact of iterating through every token of every object literal in a file. (In most cases it should probably be fine, but one can imagine a worst-case file that has many very large object literals.) The `hasComments` variable is currently declared outside all of the `if` statements here, so the comments will be enumerated in all files, not just files .

`sourceCode.commentsExistBetween` uses binary search to find comments (since it has access to all the comment locations beforehand).